migrations/2021-09-03-132838_fix_to_semver_no_prerelease/up.sql

```diff
@@ -0,0 +1,19 @@
+create or replace function to_semver_no_prerelease(text) returns semver_triple
+    immutable
+    language sql
+as
+$$
+SELECT (
+        -- 2) then, we extract the major, minor and patch numbers
+        --    (dropping the prerelease part to avoid number conversion errors)
+        split_part(version_without_metadata, '.', 1)::numeric,
+        split_part(version_without_metadata, '.', 2)::numeric,
+        split_part(split_part(version_without_metadata, '-', 1), '.', 3)::numeric
```

Comment:
The split is not necessary since the `WHERE` clause ensures that there is no '-'. Also the order in the comments is wrong because `WHERE` clauses are evaluated because `SELECT`s.

Reply:
if you try running the test suite without it you will see that it fails. it looks like the `split_part()` is evaluated even if there is a `-` in the version, and that will then cause a number conversion error because for e.g. `1.2.3-beta.4` it can't convert `3-beta.4` into a `Numeric`. at least that's what I observed when I tested this locally.